File: 0946-validate-stack-sequences/0946-validate-stack-sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        stack = []
        i = 0
        for item in pushed:
            stack.append(item)
            while(len(stack) != 0 and stack[-1] == popped[i]):
                stack.pop()
                i += 1
        return len(stack) == 0

File: 0946-validate-stack-sequences/0946-validate-stack-sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        stack<int> stack;
        int y = 0;
        for(int i = 0; i < pushed.size(); i++)
        {
            stack.push(pushed[i]);
            while(stack.size() != 0 && stack.top() == popped[y])
            {
                stack.pop();
                y++;
            }
        }
        return stack.size() == 0;
    }
};